### ISSUE TITLE
[Bug]  ekcstl upgrade Nodegroup fails when using launch templates with custom ami: asks for k8s version

### ISSUE THREAD

**Issue by joseamey** [weaveworks/eksctl#5748]
### What were you trying to accomplish?
Upgrade manage node group in aws with a launch template and launch AMIs

### What happened?
there is not need to specify the k8s version when using custom AMIs,  

getting 

`eksctl upgrade nodegroup --name=<nodegroupname less than 100 characters> --cluster=<cluster name less than 100 characters> --launch-template-version=7 --profile=<profilename> --region=<region>
`
Error: operation error EKS: UpdateNodegroupVersion, https response error StatusCode: 400, RequestID: xxx, InvalidParameterException: You cannot specify the field kubernetesVersion when using custom AMIs.

### How to reproduce it?
1. create a cluster in aws - eks , it can be 1.21, 1.22 with a launch templates and custom AMIs
2. Update the ami and create a launch template version for supporting the target upgrade version
3. Perform control plane upgrade
4. Run the ekctl upgrade Nodegroup command as I provided above. 

Provided the k8s version just in case and it returns that is not needed to specified. 

`Error: operation error EKS: UpdateNodegroupVersion, https response error StatusCode: 400, RequestID: xx InvalidParameterException: You cannot specify the field kubernetesVersion when using custom AMIs.
`**Anything else we need to know?**
aws profile

**Versions**
eksctl version
0.112.0
```
 eksctl info
eksctl version: 0.112.0
kubectl version: v1.25.1
OS: darwin
```

**Comment by danmanners** [comment]
I am also seeing this issue with Terraform. Definitely an AWS issue, but incredibly concerning.

**Comment by joseeden** [comment]
Just want to know if this is still an open issue?
We're currently using a launch template for our managed nodegroup and the launch template is pointing to the custom AMI.
The current Kubernetes version of our control plane and managed nodegroup is 1.21.

I managed to upgrade the control plane to 1.22 through the AWS Management console.
Now when I'm trying to upgrade the managed nodegroup,  don't seem to follow the steps provided in the documentation about "Updating a managed node group" (link: https://docs.aws.amazon.com/eks/latest/userguide/update-managed-node-group.html):

- Create a new version of your AMI.
- Create a new launch template version with the new AMI ID.
- Upgrade the nodes to the new version of the launch template

How are these steps upgrading the Kubernetes version of the managed node group?
I tried doing these but the managed node group always stay at version 1.21.

I have tried eksctl :
`eksctl upgrade nodegroup \
o-EKS>   --name=Demo-EKS-DMS \
>   --cluster=Demo-EKS \
>   --region=ap-southeast-1 \
>   --kubernetes-version=1.22`

But I get this error:
`Error: operation error EKS: UpdateNodegroupVersion, https response error StatusCode: 400, RequestID: 14670889-d79d-4cfc-a6bc-ad3eb4c20815, InvalidParameterException: Launch template details can't be null for Custom ami type node group`

Tried AWS CLI:
`aws eks update-nodegroup-version \
--region ap-southeast-1 \
--cluster-name Demo-EKS \
--nodegroup-name Demo \
--launch-template name=Demo-EKS,version=2 \
--kubernetes-version 1.22`

Then I get this error:
`An error occurred (InvalidParameterException) when calling the UpdateNodegroupVersion operation: You cannot specify the field kubernetesVersion when using custom AMIs.`

I'm starting to get this impression that when custom AMI is used in a managed nodegroup, it is impossible to upgrade the managed nodegroup. 

Btw, if this comment should be a separate post, I apologize. I've combed lots of pages in the internet and I can't seem to find anything. I already sent a support ticket, but I was still hoping to find something while waiting for the support's response.
Thanks.
